lab_2_tokenize_by_bpe/main.py

```diff
@@ -81,10 +189,36 @@ def decode(
     :param end_of_word_token: an end-of-word token
     :return: decoded sequence
     """
+    if not isinstance(encoded_text, list):
+        return None
+    if not isinstance(vocabulary, dict):
+        return None
+    if not isinstance(end_of_word_token, str) and end_of_word_token is not None:
+        return None
+
+    decoded_text = ''
+    for identifier in encoded_text:
+        for token in vocabulary:
```

Comment:
Consider simply inverting the dictionary

Reply:
fixed